### PR TITLE
add static delta state struct and db schema migrate

### PR DESCRIPTION
**Pull request by loadtheaccumulator**
* add static delta state struct
* add static delta state status field constants
* add db migrate for static delta state store
* add simple struct test

# Description
Add a struct to track the state of a static delta and associated Gorm DB migration

FIXES: THEEDGE-3482

Related to bug: THEEDGE-3430

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

### CODE REVIEW

**Review by ldjebran — DISMISSED** [review]
/lgtm

File: pkg/models/staticdelta_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStaticDeltaStateStruct(t *testing.T) {
+	orgID := faker.UUIDHyphenated()
+	url := faker.URL()
+	fromCommit := faker.UUIDDigit()
+	toCommit := faker.UUIDDigit()
+	name := fmt.Sprintf("%s-%s", fromCommit, toCommit)
+	status := "GENERATING"
```
Comment (neoncitylights) [inline pkg/models/staticdelta_test.go:17]: `status := StaticDeltaStatusGenerating`?

Comment (loadtheaccumulator) [reply pkg/models/staticdelta_test.go:17]: Yeah. I can do that in both places actually.

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Review by neoncitylights — APPROVED** [review]

**Pull request merged**